src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2172,40 +2172,40 @@ PeerImp::getLedger (std::shared_ptr<protocol::TMGetLedger> const& m)
             logMe += to_string (ledgerhash);
             ledger = app_.getLedgerMaster ().getLedgerByHash (ledgerhash);
 
+            if (!ledger && packet.has_ledgerseq())
+            {
+                if (auto shardStore = app_.getShardStore())
+                {
+                    auto seq = packet.ledgerseq();
+                    if (seq >= shardStore->earliestSeq())
+                        ledger = shardStore->fetchLedger(ledgerhash, seq);
+                }
+            }
+
             if (!ledger)
             {
                 JLOG(p_journal_.trace()) <<
                     "GetLedger: Don't have " << ledgerhash;
             }
+
             if (!ledger && (packet.has_querytype () &&
                 !packet.has_requestcookie ()))
             {
-                std::uint32_t seq = 0;
-                if (packet.has_ledgerseq())
+                // We don't have the requested ledger
+                // Search for a peer who might
+                auto const v = getPeerWithLedger(overlay_, ledgerhash,
+                    packet.has_ledgerseq() ? packet.ledgerseq() : 0, this);
+                if (! v || (v.get() == this))
```

Comment:
`v.get() == this` means we returned ourselves as the peer? Does that warrant its own `JLOG` message?

Reply:
It is not an error and logging it would cause superfluous messaging. It might be worth extending getPeerWithLedger to optionally exclude `this` but that beyond this PR.

Reply:
But aren't we logging now on line 2200?

Reply:
More directly, I was wondering if the existing logging message should distinguish the case of `v` being null or `v` being ourselves.

Reply:
Sorry, I misunderstood you. I am glad you brought this bit to my attention. Upon closer inspection, it is obvious the code is unnecessary and the previous enhancement I mentioned is already in place! How did we miss this? Some Self-flagellation in order for the lot!

eg.
`if (! v || (v.get() == this))` should stay as is `if (!v)`